File: app/src/main/java/com/example/petify/model/ProductModel.kt
package com.example.petify.model

import android.os.Parcelable
import com.google.gson.annotations.SerializedName

data class ProductModel(
    @SerializedName("id")
    val id: String,
    @SerializedName("supplierId")
    val supplierId: String,
    @SerializedName("price")
    val price: Int,
    @SerializedName("date")
    val date: String,
    @SerializedName("expiryDate")
    val expiryDate: String,
    @SerializedName("quantity")
    val quantity: Int,
    @SerializedName("name")
    val name: String,
    @SerializedName("image")
    val image: List<String>,
    @SerializedName("status")
    val status: String,
    @SerializedName("description")
    val description: String,
    @SerializedName("sale")
    val sale: Int
)

File: server/models/product_model.js
const db = require("../db/db");
const productSchema = new db.mongoose.Schema(
  {

    supplier_id: {
        type: db.mongoose.Schema.Types.ObjectId,
        required: true,
        ref: "supplier"
    },
    price: {
        type: Number,
        require: true
    },
    date: {
        type: String,
        required: true
    },
    expiry_Date: {
        type: String,
        required: true
    },
    quantity: {
        type: Number,
        required: true
    },
    name: {
        type: String,
        required: true
    },
    image: {
        type: Array,
        required: true
    },
    status: {
        type: String,
        required: true
    },
    description: {
        type: String,
        required: true
    },
    sale: {
        type: Number,
        required: true
    }
  },
  {
    collection: "product",
    timestamps: true, 
    versionKey: false 
  }
);
const productModel = db.mongoose.model("product", productSchema);
module.exports = { productModel };

